src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -179,11 +179,9 @@ protected void LookupMembersInternal(LookupResult result, NamespaceOrTypeSymbol
 
 #nullable enable
         protected void LookupImplicitExtensionMembersInSingleBinder(LookupResult result, Binder binder, TypeSymbol type,
-            string name, int arity, ConsList<TypeSymbol> basesBeingResolved, LookupOptions options,
+            string name, int arity, ConsList<TypeSymbol>? basesBeingResolved, LookupOptions options,
             Binder originalBinder, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
-            Debug.Assert(!type.IsTypeParameter());
```

Comment:
>Debug.Assert(!type.IsTypeParameter()); [](http://example.com/codeflow?start=12&length=38)

Not sure why this assert is removed. And, if we want to allow type parameters, then why ```isCompatible``` is not adjusted to handle them.